src/hb.hh

```diff
@@ -149,6 +149,17 @@ struct _hb_alignof
 
 #endif /* __cplusplus < 201103L */
 
+/* https://github.com/harfbuzz/harfbuzz/issues/1341 */
+#ifdef HB_NO_TYPENAME_OUTSIDE_TEMPLATE
+# ifndef typename_outside_template
+#  define typename_outside_template typename
```

Comment:
The logic seems reversed?  You want no typename here, and typename in the next?

Reply:
Also, the block is a long one to do something simple.  Basically, you are just doing:
```
#ifndef typename_outside_template
#  define typename_outside_template typename
#endif
#ifdef HB_NO_TYPENAME_OUTSIDE_TEMPLATE
#  undef typename_outside_template
#  define typename_outside_template
#endif 
```